ts/src/cryptocom.ts

```diff
@@ -1192,31 +1195,49 @@ export default class cryptocom extends Exchange {
             clientOrderId = broker + this.uuid22 ();
         }
         request['client_oid'] = clientOrderId;
+        let marketType = undefined;
+        let marginMode = undefined;
+        [ marketType, params ] = this.handleMarketTypeAndParams ('createOrder', market, params);
+        [ marginMode, params ] = this.customHandleMarginModeAndParams ('createOrder', params);
+        if ((marketType === 'margin') || (marginMode !== undefined)) {
+            request['spot_margin'] = 'MARGIN';
+        } else if (marketType === 'spot') {
+            request['spot_margin'] = 'SPOT';
+        }
+        const stopPrice = this.safeNumberN (params, [ 'stopPrice', 'triggerPrice', 'ref_price' ]);
+        if (stopPrice !== undefined) {
+            request['ref_price'] = this.priceToPrecision (symbol, stopPrice);
+        }
+        const timeInForce = this.safeStringUpper2 (params, 'timeInForce', 'time_in_force');
+        if (timeInForce !== undefined) {
+            if (timeInForce === 'GTC') {
+                request['time_in_force'] = 'GOOD_TILL_CANCEL';
+            } else if (timeInForce === 'IOC') {
+                request['time_in_force'] = 'IMMEDIATE_OR_CANCEL';
+            } else if (timeInForce === 'FOK') {
+                request['time_in_force'] = 'FILL_OR_KILL';
+            } else {
+                request['time_in_force'] = timeInForce;
+            }
+        }
         const postOnly = this.safeValue (params, 'postOnly', false);
-        if (postOnly) {
+        if ((postOnly) || (timeInForce === 'PO')) {
             request['exec_inst'] = 'POST_ONLY';
+            request['time_in_force'] = 'GOOD_TILL_CANCEL';
         }
-        params = this.omit (params, [ 'postOnly', 'clientOrderId' ]);
-        const [ marketType, marketTypeQuery ] = this.handleMarketTypeAndParams ('createOrder', market, params);
-        let method = this.getSupportedMapping (marketType, {
-            'spot': 'v2PrivatePostPrivateCreateOrder',
-            'margin': 'v2PrivatePostPrivateMarginCreateOrder',
-            'future': 'derivativesPrivatePostPrivateCreateOrder',
-            'swap': 'derivativesPrivatePostPrivateCreateOrder',
-        });
-        const [ marginMode, query ] = this.customHandleMarginModeAndParams ('createOrder', marketTypeQuery);
-        if (marginMode !== undefined) {
-            method = 'v2PrivatePostPrivateMarginCreateOrder';
-        }
-        const response = await this[method] (this.extend (request, query));
-        // {
-        //     "id": 11,
-        //     "method": "private/create-order",
-        //     "result": {
-        //       "order_id": "337843775021233500",
-        //       "client_oid": "my_order_0002"
+        params = this.omit (params, [ 'postOnly', 'clientOrderId', 'stopPrice', 'triggerPrice', 'timeInForce' ]);
+        const response = await this.v1PrivatePostPrivateCreateOrder (this.extend (request, params));
```

Comment:
I was asked to migrate this exchange to v1 because they're deprecating the other versions and v1 is a unified API that supports all markets: spot, margin, future, swap, option (warrants)